src/NServiceBus.Core/Routing/FileBasedDynamicRouting/AsyncTimer.cs

```diff
@@ -34,6 +34,10 @@ public void Start(Func<Task> callback, TimeSpan interval, Action<Exception> erro
 
         public Task Stop()
         {
+            if (tokenSource == null)
```

Comment:
I have to admit that the risk of running into that issue is very low, so this is a non-blocker for me, but I think it would still be better to have a threadsafe implementation?